iina/PlaylistViewController.swift

```diff
@@ -532,9 +532,16 @@ class PlaylistViewController: NSViewController, NSTableViewDataSource, NSTableVi
             // get related data and schedule a reload
             if Preference.bool(for: .prefetchPlaylistVideoDuration) {
               self.player.refreshCachedVideoInfo(forVideoPath: item.filename)
-              self.refreshTotalLength()
-              DispatchQueue.main.async {
-                self.playlistTableView.reloadData(forRowIndexes: IndexSet(integer: row), columnIndexes: IndexSet(integersIn: 0...1))
+              // Only schedule a reload if data was obtained and cached to avoid looping
+              if let cached = self.player.info.cachedVideoDurationAndProgress[item.filename],
+                 let duration = cached.duration {
+                if duration > 0 {
```

Comment:
This code is already heavily nested; we might as well get rid of one level.

```suggestion
              if let cached = self.player.info.cachedVideoDurationAndProgress[item.filename],
                 let duration = cached.duration,
                 duration > 0 {
```

Reply:
Fixed. Wish we had some sort of automated source code formatter. With other languages I got used to letting the IDE format and cleanup my messy code.

Reply:
This is something I would love as well, but we probably need to plan how we want to do this first :)